fairscale/optim/adascale.py

```diff
@@ -38,27 +38,36 @@
 import torch
 from torch.autograd import Variable
 import torch.distributed as dist
+from torch.optim import Optimizer
 
 
-class AdaScale(object):
+class AdaScale(Optimizer):
```

Comment:
Not sure this is needed.

In the documentation example in the rst file, things are instantiated in the following order:
```
optimizer = [base optimizer class]
scheduler = ...
optimizer = AdaScale(optimizer)
```

As long as things are done in that order, shouldn't everything work ok without needing AdaScale to subclass Optimizer?

Reply:
Looks like things happen in a different order in `test_lr_scheduler`

Reply:
I see. I meant the order in the test but I both orders are possible. Which one do you think is cleaner? It seems that if we only support a specific order, it is harder for people to understand the error when they reversed the order? What's the potential downside of subclassing from Optimizer?

Reply:
The main downside I could imagine is that we don't currently wrap or implement all methods in the Optimizer interface. If someone uses one of the methods we don't explicitly wrap or override, it wouldn't have the expected effect, since it will operate on the adascale instance rather than adascale._optimizer. I don't have a strong opinion, though. Is it strange to have a class subclass Optimizer, but then internally have an instance of Optimizer that is really providing the main implementation?

Reply:
We discussed more over chat. We will keep the subclass for now so that we maintain to be a full wrapper.